src/shared/lib/image.ts

```diff
@@ -100,3 +102,53 @@ export const compressFileImage: compressFileImage = async (file, options) => {
 
   return compressedFile;
 };
+
+export const useImageState = (source: string | { src: string }[]) => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [hasError, setHasError] = useState(false);
+  const imageCache = useRef(new Map<string, { isSuccess: boolean }>()).current;
+
+  const getImageCache = (src: string) => {
+    const imageState = imageCache.get(src);
+    return imageState ? imageState : { isSuccess: false };
+  };
+
+  useEffect(() => {
+    const loadImage = (src: string) => {
+      if (imageCache.has(src)) {
+        return Promise.resolve();
+      }
+
+      return new Promise<void>((resolve, reject) => {
+        const img = new Image();
+        img.src = src;
+        img.onload = () => {
+          imageCache.set(src, { isSuccess: true });
+          resolve();
+        };
+        img.onerror = () => {
+          imageCache.set(src, { isSuccess: false });
+          reject(new Error(`Failed to load image: ${src}`));
+        };
+      });
+    };
+
+    const loadImages = async () => {
+      try {
+        if (Array.isArray(source)) {
+          await Promise.allSettled(source.map(({ src }) => loadImage(src)));
+        } else {
+          await loadImage(source);
+        }
+        setIsLoading(false);
+      } catch {
+        setHasError(true);
+        setIsLoading(false);
+      }
+    };
```

Comment:
source가 배열인지 아닌지 확인할 필요 없이, source를 훅 내부에서 배열로 관리하는 건 어떨까요?

```suggestion
    const loadImages = async () => {
      try {
          await Promise.allSettled(source.map(({ src }) => loadImage(src)));
          
        setIsLoading(false);
      } catch {
        setHasError(true);
        setIsLoading(false);
      }
    };
```